### PR TITLE
test: migrate metabuild, metadata and net_config to snapbox

### PR DESCRIPTION
**Pull request by eth3lbert**
### What does this PR try to resolve?

Part of https://github.com/rust-lang/cargo/issues/14039.

Migrate following to snapbox:

- `tests/testsuite/metabuild.rs`
- `tests/testsuite/metadata.rs`
- `tests/testsuite/net_config.rs`

### CODE REVIEW

File: tests/testsuite/metadata.rs [hunk 1]
```diff
@@ -3503,39 +3755,43 @@ fn filter_platform() {
...
-            .replace("$NORMAL_DEP", normal_dep)
-            .replace("$FOO", &foo),
+                .replace("$ALT_TRIPLE", alt_target)
+                .replace("$ALT_DEP", alt_dep)
+                .replace("$CFG_DEP", cfg_dep)
+                .replace("$HOST_DEP", host_dep)
+                .replace("$NORMAL_DEP", normal_dep)
+                .replace("$FOO", &foo),
+            )
+            .unordered(),
```
Comment (eth3lbert) [inline tests/testsuite/metadata.rs:3773]: Ugh, this `IntoData::json(...).unordered()` just isn't working as I expected. 😅

File: tests/testsuite/metadata.rs [hunk 2]
```diff
@@ -3503,39 +3755,43 @@ fn filter_platform() {
...
 }
 "#
-            .replace("$ALT_TRIPLE", alt_target)
-            .replace("$HOST_TRIPLE", host_target)
-            .replace("$ALT_DEP", alt_dep)
-            .replace("$CFG_DEP", cfg_dep)
-            .replace("$HOST_DEP", host_dep)
-            .replace("$NORMAL_DEP", normal_dep)
-            .replace("$FOO", &foo),
+                .replace("$ALT_TRIPLE", alt_target)
```
Comment (epage) [inline tests/testsuite/metadata.rs:3766]: In theory, with snapshotting, do we need all of this manual build up?

Comment (eth3lbert) [reply tests/testsuite/metadata.rs:3766]: Yeah, I actually tried removing all the manual build parts before, but I thought that might be too aggressive. So, I decided to just carefully maintain the replacements instead.

I'll make the change.

Comment (eth3lbert) [reply tests/testsuite/metadata.rs]: Updated!
There's still an ordering issue, though.  Did I miss anything here?

File: tests/testsuite/net_config.rs [hunk 3]
```diff
@@ -67,10 +67,11 @@ fn net_retry_git_outputs_warning() {
...
     p.cargo("check -v -j 1")
         .with_status(101)
-        .with_stderr_contains(
-            "[WARNING] spurious network error \
-             (2 tries remaining): [..]",
-        )
-        .with_stderr_contains("[WARNING] spurious network error (1 tries remaining): [..]")
+        .with_stderr_data(str![[r#"
+...
+[WARNING] spurious network error (2 tries remaining): [7] Couldn't connect to server (Failed to connect to 127.0.0.1 port 11 after [..] ms: Couldn't connect to server)[..]
```
Comment (weihanglo) [inline tests/testsuite/net_config.rs:72]: Is the port number fixed?

Comment (eth3lbert) [reply tests/testsuite/net_config.rs:72]: Based on my local tests, I assume it is.  However, I'll redact it if it fails in CI.

File: tests/testsuite/metadata.rs (hunk 2 above)
Comment (eth3lbert) [reply tests/testsuite/metadata.rs]: I went ahead and reverted the json part, allowing deprecated for now. Let's move forward!

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Thank you.

@bors r+

**Pull request merged**